Signal/src/view controllers/ExperienceUpgradeViewController.swift

```diff
@@ -0,0 +1,292 @@
+        nextButton.setTitle("›", for: .normal)
+        nextButton.titleLabel?.font = UIFont.ows_mediumFont(withSize: ScaleFromIPhone5To7Plus(24, 48))
+        nextButton.addTarget(self, action:#selector(didTapNextButton), for: .touchUpInside)
+
+        // Next button layout
+        nextButton.autoPinEdge(toSuperviewEdge: .right)
+        nextButton.autoPinEdge(toSuperviewEdge: .top, withInset: arrowButtonInset)
+
+        // Dismiss button
+        let dismissButton = UIButton()
+        splashView.addSubview(dismissButton)
+        dismissButton.setTitle(NSLocalizedString("DISMISS_BUTTON_TEXT", comment: ""), for: .normal)
+        dismissButton.setTitleColor(UIColor.ows_materialBlue(), for: .normal)
+        dismissButton.isUserInteractionEnabled = true
+        dismissButton.addTarget(self, action:#selector(didTapDismissButton), for: .touchUpInside)
+
+        // Dismiss button layout
+        dismissButton.autoPinWidthToSuperview()
+        dismissButton.autoPinEdge(.top, to: .bottom, of: carouselView, withOffset: ScaleFromIPhone5(16))
+        dismissButton.autoPinEdge(toSuperviewEdge: .bottom, withInset: ScaleFromIPhone5(32))
+    }
+
+    private func buildSlideView(header: String, body: String, image: UIImage?) -> UIView {
+        Logger.debug("\(TAG) in \(#function)")
+
+        let containerView = UIView()
+        let headerView = UIView()
+        containerView.addSubview(headerView)
+        headerView.backgroundColor = UIColor.ows_materialBlue()
+
+        headerView.autoPinWidthToSuperview()
+        headerView.autoPinEdge(toSuperviewEdge: .top, withInset: -16)
+        headerView.autoSetDimension(.height, toSize: ScaleFromIPhone5(100))
+
+        // Title label
+        let titleLabel = UILabel()
+        headerView.addSubview(titleLabel)
+        titleLabel.text = header
+        titleLabel.textAlignment = .center
+        titleLabel.font = UIFont.ows_regularFont(withSize: ScaleFromIPhone5To7Plus(26, 32))
+        titleLabel.textColor = UIColor.white
+        titleLabel.minimumScaleFactor = 0.5
+        titleLabel.adjustsFontSizeToFitWidth = true;
+
+        // Title label layout
+        titleLabel.autoPinWidthToSuperview(withMargin: ScaleFromIPhone5To7Plus(16, 24))
+        titleLabel.autoPinEdge(toSuperviewEdge: .bottom, withInset: ScaleFromIPhone5(16))
+
+        let slideView = UIView()
+        containerView.addSubview(slideView)
+
+        let containerPadding = ScaleFromIPhone5To7Plus(12, 24)
+        slideView.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets(top: containerPadding, left: containerPadding, bottom: containerPadding, right: containerPadding))
+
+        // Body label
+        let bodyLabel = UILabel()
+        slideView.addSubview(bodyLabel)
+        bodyLabel.text = body
+        bodyLabel.font = UIFont.ows_lightFont(withSize: ScaleFromIPhone5To7Plus(18, 22))
+        bodyLabel.textColor = UIColor.black
+        bodyLabel.numberOfLines = 0
```

Comment:
Whenever you set `numberOfLines` to zero, you probably want: `bodyLabel.lineBreakMode = .byWordWrapping`.